### PR TITLE
Use only one MariaDB RDS instance

### PR DESCRIPTION
**Pull request by fabricetriboix**

### CODE REVIEW

File: CloudFormation/arkcase.yml [hunk 1]
```diff
@@ -550,72 +568,488 @@ Resources:
...
+    Properties:
+      DBInstanceClass: !Ref MariadbRdsInstanceClass
+      Engine: mariadb
+      EngineVersion: 10.4.8
+      AutoMinorVersionUpgrade: true
+      StorageType: gp2
+      StorageEncrypted: true
+      AllocatedStorage: !Ref MariadbRdsDiskSizeGB
+      BackupRetentionPeriod: 35
+      VPCSecurityGroups: [ !GetAtt MariadbInstanceSecurityGroup.GroupId ]
```
Comment (david-oc-miller) [inline CloudFormation/arkcase.yml:900]: Don't forget to create the MariaDB parameter group eventually.

File: LambdaFunctions/mariadb_create_database/mariadb_create_database.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        actioned = "created"
+    elif request_type == "Update":
+        sql = f"ALTER DATABASE {database_name} CHARACTER SET '{character_set}' COLLATE '{collation}';"
+        actioned = "updated"
+    else:
+        raise ValueError(f"Unknow request type: '{request_type}'")
+
+    # Get the master secret and connect to the RDS instance
+    client = boto3.client("secretsmanager")
+    value = client.get_secret_value(SecretId=secret_arn)
```
Comment (david-oc-miller) [inline LambdaFunctions/mariadb_create_database/mariadb_create_database.py:72]: We also want to create a regular database user account that will interact with this new database, right?  We should not use the master username for regular database interaction, but instead create service users like 'alfresco', 'arkcase', 'pentaho' and whatnot.

File: CloudFormation/arkcase.yml (hunk 1 above)
Comment (fabricetriboix) [reply CloudFormation/arkcase.yml:900]: @david-oc-miller Yes, that's right. Could you please remind me where I can find it?

File: LambdaFunctions/mariadb_create_database/mariadb_create_database.py (hunk 2 above)
Comment (fabricetriboix) [reply LambdaFunctions/mariadb_create_database/mariadb_create_database.py:72]: @david-oc-miller Yes, exactly. That's how it will work.

Comment (david-oc-miller) [reply LambdaFunctions/mariadb_create_database/mariadb_create_database.py:72]: You mean that's how it's working now?  I didn't see anyplace that sends a command to MariaDB that creates a user and grants access to this new schema.  Or is this something you plan on doing in the future?

File: CloudFormation/arkcase.yml (hunk 1 above)
Comment (david-oc-miller) [reply CloudFormation/arkcase.yml:900]: The parameter group I setup for EEOC is here, https://gitlab.armedia.com/infrastructure/DevOps/FedRAMP-Working/blob/master/FedRAMP-Working/mariadb_parameter_group.yaml.  Remember you need to sign onto Armedia's VPN (SonicWall app) to see gitlab.armedia.com.

File: LambdaFunctions/mariadb_create_database/mariadb_create_database.py (hunk 2 above)
Comment (fabricetriboix) [reply LambdaFunctions/mariadb_create_database/mariadb_create_database.py:72]: This above function creates a database, the `mariadb-user-secret.yml` template creates the corresponding user with password rotation etc.

Comment (david-oc-miller) [reply LambdaFunctions/mariadb_create_database/mariadb_create_database.py:72]: Well, in this function I see the CREATE DATABASE and ALTER DATABASE calls, but I don't see the SQL statements to GRANT permissions, I guess that call is somewhere else, and not affected in this MR?  I don't remember where it is.

Comment (fabricetriboix) [reply LambdaFunctions/mariadb_create_database/mariadb_create_database.py:72]: @david-oc-miller The `alfresco` database user will be created by the rotation function. When the secret rotation schedule is created by CloudFormation, SecretsManager will immediately rotate the secret, which will create the user. I guess this is not evident, so I will document this in the code.

**Pull request merged**